Modules/_sqlite/row.c

```diff
@@ -151,9 +150,13 @@ pysqlite_row_subscript(PyObject *op, PyObject *idx)
         PyObject *item = PyTuple_GetItem(self->data, _idx);
         return Py_XNewRef(item);
     } else if (PyUnicode_Check(idx)) {
-        nitems = PyTuple_Size(self->description);
+        if (Py_IsNone(self->description)) {
+            PyErr_Format(PyExc_IndexError, "No item with key %R", idx);
```

Comment:
IMO, this should be a `KeyError`, but that would create an inconsistency with L174. Moreover, we should also improve that error message by aligning it to this.

Reply:
IMO it should have been a dedicated exception type inheriting from LookupError but the ship has already sailed.

Now, since using the named field instead of its position is equivalent to `row[get_index(field_name)]`, I think it's fine to keep the IndexError (even though it could be surprising at first).

Improving the error message may be good or documenting that an IndexError is raised for invalid indices or invalid keys (if it's not already done).

Reply:
>  I think it's fine to keep the IndexError (even though it could be surprising at first).

Definitely.

> Improving the error message may be good or documenting that an IndexError is raised for invalid indices or invalid keys (if it's not already done).

Improving the error message is out of scope for this PR (such a change should not be backported). A docs update could also be nice, but I don't have the bandwidth for that right now.